meson.build

```diff
@@ -218,6 +244,10 @@ if host_os_family == 'linux' or host_os_family == 'qnx'
   if not have_libelf
     error('libelf is required')
   endif
+  if not have_frida_glib
+    elf_dep = dependency('libelf')
+    extra_deps += [elf_dep]
+  endif
```

Comment:
This should be done unconditionally inside this branch – but I seem to recall that the reason we check for `libelf.h` here and manually add `-lelf` is that a `.pc` is not available on all systems (e.g. Android and QNX)... So I wonder if we can make the existing logic work for GIR somehow? Like setting `required: false`, and taking the `-lelf` route as a fallback.

(Do we actually need libelf for GIR now that we're skipping backend bits?)

Reply:
We are only skipping backend bits for GIR generation, not for linking Gum, so libelf is required to link still. The `.pc` for libelf is _not_ provided in the Frida toolchain, which is why we cannot do a `dependency` in here. However, on a non-Frida-GLib build we are targeting the system GLib, so we must use `dependency` to get `libelf`. I am not clear on how using `required: false` and setting a fallback is any better than this?